packages/proxygen/tests/checkRequiredKeys.test.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable  max-len */
+
+import {checkRequiredKeys, Proxygen} from "../src/helpers"
+
+// Mock data
+const validProxygen: Proxygen = {
+  apiName: "testApi",
+  proxygenSecretName: "testSecret",
+  kid: "testKid"
+}
+
+describe("checkRequiredKeys", () => {
+  it("should not throw an error if all default required keys are present", () => {
+    expect(() => checkRequiredKeys(validProxygen, [])).not.toThrow()
+  })
+
+  it("should throw an error if a required key is missing", () => {
+    const requiredKeys = ["missingKey"]
+    expect(() => checkRequiredKeys(validProxygen, requiredKeys)).toThrow(
+      "Input is one of missing required keys: apiName,proxygenSecretName,kid,missingKey. Input keys: apiName,proxygenSecretName,kid"
+    )
+  })
+
+  it("should throw an error if obj is empty", () => {
+    expect(() => checkRequiredKeys({} as Proxygen, [])).toThrow(
+      "Input is one of missing required keys: apiName,proxygenSecretName,kid. Input keys: "
+    )
+  })
+
+  it("should not throw an error if obj has extra non required keys", () => {
+    const objWithExtraKeys = {
+      ...validProxygen,
+      extraKey: "extraValue"
+    }
+    expect(() => checkRequiredKeys(objWithExtraKeys, [])).not.toThrow()
+  })
+
+  it("should not throw an error if obj has extra required keys", () => {
+    const requiredKeys = ["environment"]
+    const objWithExtraKeys = {
+      ...validProxygen,
+      environment: "extraValue"
+    }
+    expect(() => checkRequiredKeys(objWithExtraKeys, requiredKeys)).not.toThrow()
+  })
+
+  it("should throw an error if obj has missing required default keys", () => {
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const objWithMissingKeys: any = {
```

Comment:
Do we need this explicit any? Can't we just let the type inference do its thing?

Reply:
no - because its testing a missing required key. If you leave it blank, then it complains when you try to delete a required key